rs_state_test.go

```diff
@@ -312,3 +313,22 @@ func TestNewReplicaSetStateFailure(t *testing.T) {
 		t.Fatalf("unexpected error: %s", err)
 	}
 }
+
+func TestReplicaStateFailsFast(t *testing.T) {
```

Comment:
this is non deterministic right, because if running in parallel we could pause between start time and stop time, and therefore exceed the acceptable latency.